math/gfm/math/wideint.d

```diff
@@ -114,6 +114,80 @@ struct wideIntImpl(bool signed, int bits)
         opAssign!T(x);
     }
 
+    /// Construct from compile-time digit string.
+    ///
+    /// Both decimal and hex digit strings are supported.
+    ///
+    /// Example:
+    /// ----
+    /// auto x = int128.literal!"20_000_000_000_000_000_001";
+    /// assert((x >>> 1) == 0x8AC7_2304_89E8_0000);
+    ///
+    /// auto y = int126.literal!"0x1_158E_4609_13D0_0001";
+    /// assert(y == x);
+    /// ----
+    template literal(string digits)
+    {
+        static bool isValidDigitString(string digits)
```

Comment:
These should be moved outside the template, otherwise there'll be one `isValidDigitString` function and one `impl` function per unique literal.